Issue 161: by column with plot_grid

BrianLang opened the pull request:
Fix #161 
- Added argument bycol to function call.
- Calculate plot ordering if bycol is TRUE.
- Moved grobs calculation to after plots ordering.

clauswilke commented:
I'm fine with this in principle, but this PR will need a lot of work before it can be merged. Just off the top of my head:

- Call the argument `byrow` as in `matrix()`.

- Place the new argument before the deprecated arguments.

- Don't use `F`, write `FALSE`.

- The new argument needs to be documented and documentation re-generated.

- There needs to be an entry in the changelog (`NEWS.md`). Look at the ggplot2 source for how to format contribution by outside contributors.

- There needs to be a visual test for the new feature. Visual tests are currently failing, so I need to fix them first. I suggest you develop the code for the visual test but just post it here in a comment rather than adding to the code at this time, so we can discuss.

BrianLang commented:
Thanks for this feedback, I now have done most of these things to the extent that I understand what you want. 
- I've updated the argument to be `byrow`.
- I've moved the argument to be before `rows` and `cols`.
- I've used `FALSE` instead of `F`  
- I've written argument documentation and an example and re-generated the documentation.  
- I added this to the NEWS.md:
>> ## Minor changes
>>- `plot_grid()` now accepts `byrow` as an argument to place plots by row or by column onto the plot area.  (@BrianLang, #162).  
- For a visual test I think we would add this to the `test_that("basic plot arranging works"` test:
```
expect_doppelganger("colwise arranging",
                      plot_grid(p1, NULL, p2, NULL, p3, NULL, byrow = FALSE) + theme_map()
  )
```

clauswilke commented on R/plot_grid.R:
```diff
@@ -96,6 +97,13 @@
+#' plot_grid(
+#' p1, NULL, p2, NULL, p3, NULL,
+#' ncol = 2,
+#' byrow = TRUE
```
Please indent the three argument lines by 2 spaces.
